CMakeLists.txt

```diff
@@ -532,6 +530,82 @@ while(library_variants_copy)
             ${INSTALL_DIR}/lib/${target} ${install_directory}/lib
     )
 
+    if(libc_type STREQUAL "picolibc")
+        ExternalProject_Add(
```

Comment:
It seems you're adding a second picolibc build with tests enabled. I don't like it, particularly as building picolibc takes a non-trivial amount of time.

The picolibc tests seem very simple. If there isn't a clean way to build the tests with meson then I'd be OK with building them with CMake, even if that means duplicating a lot of https://github.com/picolibc/picolibc/blob/main/test/CMakeLists.txt

Reply:
Done.